### PR TITLE
SCT-626 Refactor elastic client usage

### PR DESCRIPTION
**Pull request by MrCreosote**
All client usage now occurs in one method.

### CODE REVIEW

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java [hunk 1]
```diff
@@ -1691,25 +1686,44 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder hacb) {
...
     
-    public Response makeRequest(String reqType, String urlPath, Map<String, ?> doc, 
-            Map<String, String> attributes) throws IOException {
+    private Response makeRequest(
+            final String reqType,
+            final String urlPath,
+            final Map<String, ?> doc, 
+            final Map<String, String> attributes)
+            throws IOException {
+        return makeRequest(reqType, urlPath, attributes, doc == null ? null : stringEntity(
```
Comment (arfathpasha) [inline lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: (HttpEntity)stringEntity(
                UObject.transformObjectToString(doc))

Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: Why is the cast necessary?

**Comment by MrCreosote** [comment]
I'm going to merge since I'm ready for the next PR.

File: lib/src/kbasesearchengine/search/ElasticIndexingStorage.java (hunk 1 above)
Comment (arfathpasha) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: Helps the reader match the arguments to the right function signature.

Comment (MrCreosote) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: So you're advocating for including a cast every time a polymorphic method is called and the type is unclear?

Comment (arfathpasha) [reply lib/src/kbasesearchengine/search/ElasticIndexingStorage.java]: My mistake. My IDE was not making the connection to the function signature and it was an error with the way I manually included the jars. Please ignore.

**Pull request merged**